backend/typescript/utilities/stripeUtils.ts

```diff
@@ -90,6 +92,83 @@ export async function createStripePrice(
   return priceObject;
 }
 
+export const createStripeLineItems = (
+  sessionsToRegister: CampSession[],
+  campers: CreateCampersDTO,
+  dropoffPriceId: string,
+  pickupPriceId: string,
+): Stripe.Checkout.SessionCreateParams.LineItem[] => {
+  return sessionsToRegister.flatMap((campSession) => {
+    const priceItems = [
+      { price: campSession.campPriceId, quantity: campers.length },
+    ];
+
+    // every camper belongs to every session
+    // all same EDLP price
+    // campers have different EDLP dates, which may belong to different sessions
+    // we want to isolate EDLP per session
+
+    // Assumes that campers are only registered for one session per day
+    const [earlyDropoffs, latePickups] = campers.reduce(
+      ([earlyTotal, lateTotal]: [number, number], camper) => {
+        let camperEarlyTotal = 0;
+        if (camper.earlyDropoff) {
+          const camperEDDates = new Set(camper.earlyDropoff);
+          const edDates = campSession.dates.filter((date) =>
+            camperEDDates.has(date.toString()),
+          );
+
+          camperEarlyTotal += edDates.length;
+        }
+
+        let camperLateTotal = 0;
+        if (camper.latePickup) {
+          const camperLPDates = new Set(camper.latePickup);
+          const lpDates = campSession.dates.filter((date) =>
+            camperLPDates.has(date.toString()),
+          );
+          camperLateTotal += lpDates.length;
+        }
+
+        return [earlyTotal + camperEarlyTotal, lateTotal + camperLateTotal];
+      },
+      [0, 0],
+    );
```

Comment:
this is wrong -- identifies # of days, not # of EDLP slots. correct in later PR

Reply:
If we only care about the number of EDLP slots, then we can always just compare against the camp start and end time right? All sessions in the camp will start and end at the same time. So we might not care about which session the EDLP slot belongs to if we can just use the start and end time to calculate the number of slots for ED and LP. 

Also the camper model has 2 separate array fields for ED and LP so you could use those to identify if you need start or end time

Reply:
nvm I think you did that in the next PR

Reply:
yeah I need another PR to fix a lot of the EDLP logic